### PR TITLE
Feat model transformer node

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* Include a summary of the changes and the related issue. 
* Include relevant motivation and context.

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/workflow/ops/model-transformer/model-transformer-operation.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Operation, WorkflowOperationTypes } from '@/types/workflow';
+
+interface ModelOperationState {
+	modelId: string | null;
+	modelConfigurationIds: any[];
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-transformer/model-transformer-operation.ts:5]: Configuration ids are  `strings[]`

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
+<template>
+	<div class="background">
+		<Suspense>
+			<tera-model-jupyter-panel
+				asset-id="bcd32c48-1daa-4a52-8ed7-147c1dc4b0b0"
+				:model-configuration-id="modelConfigurationId"
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:6]: Why do we have a hardwired `bcd32c48-1daa-4a52-8ed7-147c1dc4b0b0`

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+			data: { history: [] },
+			timestamp: new Date().toISOString()
+		});
+		notebookSessionId = response?.id;
+
+		if (notebookSessionId) {
+			// update the node state with the notebook session id
+			const state = cloneDeep(props.node.state);
+			state.notebookSessionId = notebookSessionId;
+			workflowEventBus.emit('update-state', {
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:65]: Your state setup is strange - I know that the typing information is lost after passing it through a prop. But if we type it out:

```
const state = cloneDeep(props.node.state) as ModelTransformState; 
```

And we don't even have a `ModelTransformState` here, neither is the notion of `state.notebookSessionId`

Incidentally if you know a good way to enforce types please let me know :)

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	const state = cloneDeep(props.node.state);
+	state.modelId = model?.id;
+	// update node state with the model id
+	workflowEventBus.emit('update-state', { node: props.node, state });
+
+	// set default configuration
+	await addDefaultConfiguration(model);
+
+	// setting timeout...elastic search might not update default config in time
+	await setTimeout(async () => {
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:91]: This should be fixed for document creation in ES, can we verify that this is working without the timeout?

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer-node.vue [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+const fetchModel = async () => {
+	if (!props.node?.state?.modelId) return;
+	model.value = await getModel(props.node?.state?.modelId);
+};
+
+onMounted(async () => {
+	await fetchModel();
+});
+watch(
+	() => props.node.state,
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer-node.vue:31]: You may want to tighten this watcher a bit if the only thing you need is the modelId field,

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:6]: sorry i believethe asset-id is not longer used and I was using it for testing

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 4 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:91]: hmm - ill test again, I was having issues where I wasn't getting the default config back. Will test without

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 3 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:65]: fair, I hadn't used the state type at all and will get into the habit of using it.  Change will be made.  

Im not too sure about enforcing types, will have to be something I can look into

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 4 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:91]: did a few tests, unfortunately it looks like the timeout is needed!

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer-node.vue (hunk 5 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer-node.vue:31]: done!

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 3 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:65]: i've typed the states, I've also done the some with the dataset transformer

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue:5]: this has been removed...

File: packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue (hunk 4 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue]: okay ... we will come back to this later

**Review by mwdchang — APPROVED** [review]

**Pull request merged**